packages/mosaic-tanstack-table-core/src/view-model.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Base class for View Models that manage Mosaic selections, topology, and schema mapping
+ * independent of the UI framework. Provides lifecycle management for listeners and clients.
+ */
+
+import type { Coordinator, MosaicClient, Selection } from '@uwdata/mosaic-core';
+import type { MosaicDataTableColumnDefMetaOptions } from './types';
+
+export abstract class MosaicViewModel {
```

Comment:
Do not use abstract classes. What this means is that the user would need to redefine/recreate a class on the user's side which satisfies the types of the abstract class.

Use a regular class with callbacks which can be setup. You can even make it simpler after the fact by wrapping it in a factory function. You've already done something similar for this with `toggleMosaicSelection`.